app/services/user.js

```diff
@@ -0,0 +1,20 @@
+const bcrypt = require('bcryptjs');
+const { User } = require('../models');
+const errors = require('../errors');
+const logger = require('../logger');
+
+exports.hashPassword = password => bcrypt.hash(password, 10);
+
+exports.checkEmailExist = ({ email }) =>
+  User.findOne({ where: { email } }).catch(err => {
+    logger.error('An error occurred in data base');
+    throw errors.databaseError(err.message);
```

Comment:
this is a database error but is a specific error, is because a constraint is being violated